ts/ui/menu/Menu.ts

```diff
@@ -507,11 +507,20 @@ export class Menu {
         this.a11yVar<boolean>('viewBraille'),
         this.a11yVar<boolean>('voicing'),
         this.a11yVar<string>('locale', locale => this.setLocale(locale)),
-        this.a11yVar<string>('speechRules', value => {
-          const [domain, style] = value.split('-');
-          this.document.options.sre.domain = domain;
-          this.document.options.sre.style = style;
-        }),
+        {
+          name: 'speechRules',
+          getter: () => {
+            return this.settings['speechRules'];
+          },
+          setter: (value: string) => {
+            const [domain, style] = value.split('-');
+            this.settings['speechRules'] = value;
+            this.document.options.sre.domain = domain;
+            this.document.options.sre.style = style;
+            this.rerender(STATE.COMPILED);
+            this.saveUserSettings();
+          }
+        },
```

Comment:
You should be able to use `this.variable()` rather than `this.a11yvar()` to define this, so you don't have to do the `getter` and setter by hand (and won't need to include lines 517 and 521, which are part of what `this.variable()` does.  So

``` js
          this.variable<string>('speechRules', value => {
            const [domain, style] = value.split('-');
            this.document.options.sre.domain = domain;
            this.document.options.sre.style = style;
            this.rerender(STATE.COMPILED);
          }),
```

should do it (that is, changing `a11yvar` to `variable` and adding the `rerender()` call to the original lines 510 to 514 should be enough).

Reply:
I tried that and it not work. And it drove me crazy, so I decided to stick with the pedestrian version. 
But I will try again. Maybe I was just too thick the other day.

Reply:
I just tried this out, and it seemed to work fine for me.  Give it another try, and see if you have any problems.